### DISCUSSION TITLE
How is React Select focusing the input everytime we make a selection?

### DISCUSSION THREAD

**Discussion by florinpavel22** [JedWatson/react-select#5276] [Q&A] [answered]
Hi guys, if we give react-select `closeMenuOnSelect={false}` prop, if I select an option, the input stays focused, allowing me to type and filter the options.
How is this being done? Any direction to where I should look?

Sample: https://codesandbox.io/s/pedantic-mclaren-b53bwb

**Comment by manjushsh** [comment, marked as answer]
@florinpavel22 

[Input has ref. So focus is set through ref here](https://github.com/JedWatson/react-select/blob/4b8468636bcfadf0cfe45f9a7a6c1db5dca08d9a/packages/react-select/src/Select.tsx#L773)

[On Mouse Down at line 1151, which sets focus](https://github.com/JedWatson/react-select/blob/4b8468636bcfadf0cfe45f9a7a6c1db5dca08d9a/packages/react-select/src/Select.tsx#L1145)

[On Click of menu item, onMouseDown is called @ Line 1912](https://github.com/JedWatson/react-select/blob/4b8468636bcfadf0cfe45f9a7a6c1db5dca08d9a/packages/react-select/src/Select.tsx#L1903)

**Reply by florinpavel22** [reply]
Perfect, thank you @manjushsh!
